agenda-webapps/src/main/webapp/vue-app/agenda-common/components/event/view/AgendaEventIcs.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <div>
+    <div class="d-flex justify-center">
+      <v-btn outlined color="primary" class="btn border-radius v-btn v-btn--contained v-size--default v-chip v-chip--outlined theme--light
+         primary primary--text mt-4 mb-2 me-5" @click="downloadICS">
+        <v-icon size="20" class="uiIcon24x24 pe-2" depressed>
+          fa-calendar-plus
+        </v-icon>
+        {{ $t('agenda.icsbutton') }}
+      </v-btn>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    settings: {
+      type: Object,
+      default: () => null
+    },
+    event: {
+      type: Object,
+      default: () => ({})
+    },
+  },
+  methods: {
+    generateICS(event) {
+      const formatDate = (date) => {
+        return date ? `${new Date(date).toISOString().replace(/[-:]/g, '').split('.')[0]}Z` : '';
+      };
+
+      const foldLine = (line) => {
+        const maxLength = 70;
+        if (line.length <= maxLength) { return line; }
+        let result = '';
+        while (line.length > maxLength) {
+          result += `${line.substring(0, maxLength)}\r\n `;
+          line = line.substring(maxLength);
+        }
+        return result + line;
+      };
+
+      const htmlDescription = `Invitation transmise par <b>${this.event.creator.dataEntity.profile.lastname} ${this.event.creator.dataEntity.profile.firstname}</b> dans l'espace <b>${this.event.calendar.title}</b>.
+      ${event.conferences[0].url ? `<br><b>Lien de visio</b> : <a href="${event.conferences[0].url}">${event.conferences[0].url}</a><br>` : ''}
+      ${event.description ? `<br><b>Détail de l'évènement :</b><br>${event.description}` : ''}
+      `.replace(/\\/g, '\\\\').replace(/;/g, '\\;').replace(/,/g, '\\,').replace(/\n/g, '\\n');
+
+      const icsContent = 'BEGIN:VCALENDAR\r\n' +
+        'CALSCALE:GREGORIAN\r\n' +
+        'METHOD:PUBLISH\r\n' +
+        'PRODID:-//eXo Platform//eXo calendar//EN\r\n' +
+        'VERSION:2.0\r\n' +
+        'BEGIN:VEVENT\r\n' +
+        `UID:${event.id || ''}\r\n` +
+        `DTSTAMP:${formatDate(new Date())}\r\n` +
+        `DTSTART:${formatDate(event.startDate)}\r\n` +
+        `DTEND:${formatDate(event.endDate)}\r\n` +
+        `SUMMARY:${event.summary || ''}\r\n` +
+        `DESCRIPTION:${htmlDescription || ''}\r\n` +
+        `X-ALT-DESC;FMTTYPE=text/html:${htmlDescription}\r\n` +
+        `LOCATION:${event.location || ''}\r\n` +
+        `URL:${event.conferences[0].url || ''}\r\n` +
+        `ORGANIZER;CN=${this.event.creator.dataEntity.profile.lastname} ${this.event.creator.dataEntity.profile.firstname}:MAILTO:${this.event.creator.dataEntity.profile.email}\r\n` +
+        'END:VEVENT\r\n' +
+        'END:VCALENDAR\r\n';
+      return icsContent.split('\r\n').map(foldLine).join('\r\n');
+    },
+    downloadICS() {
+      const event = this.event;
+      const icsContent = this.generateICS(event);
```

Comment:
Could this simplified like this ?
```suggestion
      const icsContent = this.generateICS(this.event);
```